digicampipe/scripts/pulse_shape.py

```diff
@@ -3,63 +3,211 @@
 Reconstruct the pulse template
 
 Usage:
-  pulse_shape.py [options] [FILE] [INPUT ...]
+  digicam-template [options] <outfile_path> <input_files>...
 
 Options:
   -h --help               Show this screen.
-  --max_events=N          Maximum number of events to analyse
-  -o FILE --output=FILE.  Output file.
-  -i INPUT --input=INPUT. Input files.
-  -c --compute            Compute the data.
-  -f --fit                Fit.
-  -d --display            Display.
-  -v --debug              Enter the debug mode.
-  -p --pixel=<PIXEL>      Give a list of pixel IDs.
+  --plots=PATH            path where to save a pdf full of interesting images.
+  --chunk_size INT        how many events to read before histogramming.
+                          This uses about 1GB memory, if you have less mem,
+                          reduce this number (will make the program slower).
+                          [default: 10000]
 '''
-import os
-from docopt import docopt
-from tqdm import tqdm
-from schema import Schema, And, Or, Use, SchemaError
 import numpy as np
-import matplotlib.pyplot as plt
-
+import h5py
 from digicampipe.io.event_stream import calibration_event_stream
-from digicampipe.io.containers_calib import CalibrationContainer
+
+from tqdm import tqdm
+from docopt import docopt
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+
+from ctapipe.visualization.mpl import CameraDisplay
+from digicampipe.utils import DigiCam
+
+
+def fill_histo_from_buffer(histo, buffer):
```

Comment:
maybe we can replace this by the `histogram` package. I created it to avoid having to do chunks. I personally think chunks add complexity to the code and are more error prone
As you prefer

Reply:
Ah well maybe I am wrong because you have histogram bin counts of shape = (n_pixels, n_samples)

Reply:
This needs a 2D histogram, the histogram package at the moment seems to support only 1D histograms